### ISSUE TITLE
add a tutorial for decision trees, random forests and gradient boosting methods

### ISSUE THREAD

**Issue by kmike** [TeamHG-Memex/eli5#119]
I think sklearn.tree, sklearn.ensemble and xgboost support deserves its own tutorial. It should show examples for all of them and explain how to interpret the results:

* what do feature importances mean (in case of sklearn: what's Gini Importance; see also: https://github.com/glouppe/phd-thesis);
* how to interpret graphviz and text output for a single decision tree;
* how we explain xgboost, RandomForest.., Gradientboosting..., ExtraTrees... predictions.

**Comment by ritviksahajpal** [comment]
thanks for an excellent package @kmike , are you planning to add a tutorial for a regression example as well?

**Comment by kmike** [comment]
A good call @ritviksahajpal, it'd be great for tutorial for cover regression as well. Thanks for the suggestion!

**Comment by lopuhin** [comment]
I started writing a tutorial on https://www.kaggle.com/c/titanic/data - it has numeric, categorical and text features, is small and easy to understand. The plan is to mostly use XGBoost cause it seems more popular, but also show that it works with sklearn tree ensembles as well.

**Comment by kmike** [comment]
Fixed by https://github.com/TeamHG-Memex/eli5/pull/136. I opened another ticket (#145) for the regression tutorial.
